drivers/pwm/pwm_nrfx.c

```diff
@@ -248,6 +284,20 @@ static int pwm_nrfx_set_cycles(const struct device *dev, uint32_t channel,
 		 * until another playback is requested (new values will be
 		 * loaded then) or the PWM peripheral is stopped.
 		 */
+#if PWM_NRFX_USE_CLOCK_CONTROL
+		if (!data->clock_requested) {
+			int ret = nrf_clock_control_request_sync(config->clk_dev,
```

Comment:
This is only needed for the fast instances. You could have `clk_dev` set to NULL for others and avoid this call.